src/main/java/org/zendesk/client/v2/DefaultUserAgent.java

```diff
@@ -0,0 +1,39 @@
+package org.zendesk.client.v2;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+public class DefaultUserAgent {
```

Comment:
I'm not against this change, but it's unrelated to the migration to CBP.
This PR is already big enough, I'd rather see this in a different one.
Two notes (that we discuss later in a the split PR):
* end user can easily change this so you're not guaranteed to get the information server side.
* AFAIR the jar manifest should already contain the artifact version, it might be simpler to pick it up this way.